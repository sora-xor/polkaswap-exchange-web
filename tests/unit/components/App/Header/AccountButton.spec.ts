// import Vuex from 'vuex';
// import { shallowMount } from '@vue/test-utils';
//
// import AccountButton from '@/components/App/Header/AccountButton.vue';
// import { Account } from '@soramitsu/soraneo-wallet-web/lib/types/common';
// import Theme from '@soramitsu-ui/ui-vue2/lib/types/Theme';
// import { useDescribe, localVue } from '../../../../utils';
// import { MOCK_ACCOUNT } from '../../../../mocks/account';
// import { formatAddress } from '../../../../../src/utils';
//
// const createStore = (loggedIn: boolean, mockAccount = MOCK_ACCOUNT) =>
//   new Vuex.Store({
//     getters: {
//       isLoggedIn: () => loggedIn as boolean,
//       account: () => mockAccount as Account,
//       libraryTheme: () => Theme.LIGHT as string,
//     },
//   });
//
// useDescribe('AccountButton.vue', AccountButton, () => {
//   it('should be rendered correctly', () => {
//     const wrapper = shallowMount(AccountButton, {
//       localVue,
//       store: createStore(true),
//     });
//
//     expect(wrapper.element).toMatchSnapshot();
//   });
//
//   it('should show connect account text when not logged in', () => {
//     const wrapper = shallowMount(AccountButton, { localVue, store: createStore(false) });
//     const div = wrapper.find('.account-control-title');
//
//     expect(div.text()).toMatch('Connect account');
//   });
//
//   it('should show user name when logged in', () => {
//     const wrapper = shallowMount(AccountButton, { localVue, store: createStore(true) });
//     const div = wrapper.find('.account-control-title');
//
//     expect(div.text()).toMatch(MOCK_ACCOUNT.name);
//   });
//
//   it('should show address when name is not provided', () => {
//     const mockAccount = {
//       ...MOCK_ACCOUNT,
//       name: '',
//     };
//     const wrapper = shallowMount(AccountButton, { localVue, store: createStore(true, mockAccount) });
//     const div = wrapper.find('.account-control-title');
//
//     expect(div.text()).toMatch(formatAddress(MOCK_ACCOUNT.address, 8));
//   });
// });

describe('Test', () => {
  test('Temporary test', () => {
    expect(true).toEqual(true);
  });
});
