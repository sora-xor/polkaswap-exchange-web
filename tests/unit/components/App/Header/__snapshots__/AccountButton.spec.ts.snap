// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountButton.vue should be rendered correctly 1`] = `
<s-button-stub
  borderradius="small"
  class="account-control s-pressed"
  icon=""
  iconposition="left"
  nativetype="button"
  size="medium"
  tabindex="0"
  tooltip="Connected account"
  tooltipplacement="top"
  type="tertiary"
  usedesignsystem="true"
>
  <div
    class="account-control-icon"
  >
    <walletavatar-stub
      address="cnRXua6zs8TaE87BQFL6uWVbT2g6GXsUjwk6PTvL6UHcHDCvo"
      size="32"
      theme="polkadot"
    />
  </div>
   
  <div
    class="account-control-title name"
  >
    Sora Name
  </div>
</s-button-stub>
`;
